src/jit/compile.c

```diff
@@ -1,83 +1,120 @@
 #include "moar.h"
-#include "dasm_proto.h"
+#include "internal.h"
 #include "platform/mmap.h"
-#include "emit.h"
 
-#define COPY_ARRAY(a, n, t) memcpy(MVM_malloc(n * sizeof(t)), a, n * sizeof(t))
+
+void MVM_jit_compiler_init(MVMThreadContext *tc, MVMJitCompiler *compiler, MVMJitGraph *jg);
+void MVM_jit_compiler_deinit(MVMThreadContext *tc, MVMJitCompiler *compiler);
+MVMJitCode * MVM_jit_compiler_assemble(MVMThreadContext *tc, MVMJitCompiler *compiler, MVMJitGraph *jg);
+
+#define COPY_ARRAY(a, n) memcpy(MVM_malloc(n * sizeof(a[0])), a, n * sizeof(a[0]))
```

Comment:
This would be preferable as a `MVM_STATIC_INLINE` rather than a macro.

Reply:
In that case it couldn't 'automatically' take over the types, which the macro can; it's really just a convenience though, so I don't care that much.

Reply:
Ah, good point. Can stay as a macro then.